go-present-mode.el

```diff
@@ -0,0 +1,1278 @@
+;; with Go present tool web view.
+;;
+;;
+;; Installation
+;; ------------
+;; (add-to-list 'load-path "path/to/go-present-mode/")
+;; (require 'go-present-mode)
+;;
+;;
+;; Installation with use-package
+;; -----------------------------
+;; (use-package go-present-mode
+;;   :load-path "path/to/go-present-mode/")
+;;
+;;
+;; Features
+;; --------
+;;
+;; Standard Emacs mode features such as: syntax highlighting via
+;; font-lock, commenting/uncommeting, imenu support, and defun
+;; movement.
+;;
+;; Outline-mode integration:
+;;    C-c C-o C-h         collapse all headlines
+;;    C-c C-o C-s         show all headlines
+;;    C-c C-o C-n         go to next headline
+;;    C-c C-o C-p         go to previous headline
+;;
+;; Present and allow editing (C-c C-p C-l) links.
+;;
+;; Open slide at point in a web browser (C-c C-p C-p).
+;;
+;; Show inline images (toggle with C-c C-p C-i).
+;;
+;; Open thing at point (C-c C-p C-o) such as: image, code, link.
+;;
+;; Toggle fonts
+;;   C-c C-p C-`         program
+;;   C-c C-p C-*         bold
+;;   C-c C-p C-_         italic
+;;
+;; Show preformatted text as such.
+;;
+;; Show inline code (C-c C-p C-v) as specified by .code
+;; commands. Support:
+;;   - Address handling, OMIT lines, line numbers, and highlights.
+;;   - Modifying code inline.
+;;   - Executing code at point (C-c C-p C-c).
+;;   - Tracking changes to the underlying file.
+
+;;; Code:
+
+(require 'cl-lib)
+(require 'outline)
+(require 'overlay)
+
+(defgroup go-present nil
+  "Configuration for go-present-mode."
+  :group 'go)
+
+(defvar go-present-mode-syntax-table
+  (let ((tbl (make-syntax-table)))
+    (dolist (ch (string-to-list "'\"~!@#$%^&-+=<>|"))
+      (modify-syntax-entry ch "." tbl))
+    (dolist (ch (string-to-list "_*`"))
+      (modify-syntax-entry ch "_" tbl))
+    (modify-syntax-entry ?# "<" tbl)
+    (modify-syntax-entry ?\n ">" tbl)
+    tbl)
+  "Syntax table for `go-present-mode'.")
+
+(defun go-present--propertize-syntax (start end)
+  "Fix the syntax table in code sections.
+
+Mark # between START and END as punctuation because there are no
+comments in preformatted sections."
+  (save-excursion
+    (goto-char start)
+    (while (search-forward "#" end t)
+      (when (get-text-property (point) 'go-present--code)
+        (put-text-property (1- (point)) (point)
+                           'syntax-table '(1))))))
+
+(defcustom go-present-max-image-size 500
+  "Maximum size of image dimensions.
+If neither dimension is specified, then limit width.  Otherwise
+limit the larger one."
+  :type 'integer
+  :group 'go-present)
+
+(defcustom go-present-base-port 4000
+  "Base port to serve Go present on.
+
+Go present serves on the first free port that is not smaller than
+`go-present-base-port'."
+  :type 'integer
+  :group 'go-present)
+
+(defface go-present-bold-face '((t (:inherit bold)))
+  "Face for bold text outside code sections.")
+(defvar go-present-bold-face 'go-present-bold-face)
```

Comment:
It doesn't work without defvars. When I wrote the code, I spent some time researching this and couldn't figure out how to implement fontification without defvars in this case (if you comment out defvars then nothing is fontified). In the end I decided that it wasn't worth the time to look into this further.

If you know how to fix it, then I'm happy to update the code. In fact, I would really appreciate the feedback because I wasted so much time on this (and I hope that it wasn't me just missing something obvious).

I suspect that I'd have to update references to those faces in go-present--keywords definition, but I'm not sure how.